server/app/views/applicant/NorthStarAnswerData.java

```diff
@@ -62,6 +64,24 @@ private ImmutableList<String> fileNames() {
     return ImmutableList.copyOf(fileNames);
   }
 
+  public ImmutableList<String> urls() {
+    ArrayList<String> urls = new ArrayList<String>();
+
+    AnswerData data = this.answerData;
+    if (!data.encodedFileKeys().isEmpty()) {
+      for (int i = 0; i < data.encodedFileKeys().size(); i++) {
+        String encodedFileKey = data.encodedFileKeys().get(i);
+        String fileUrl = controllers.routes.FileController.show(applicantId, encodedFileKey).url();
+        urls.add(fileUrl);
+      }
+    } else if (data.encodedFileKey().isPresent()) {
```

Comment:
**question:** Will this single encoded file key be removed once multi-file upload is fully launched? If so, could we maybe add a TODO to remove it with https://github.com/civiform/civiform/issues/7493 included?

Reply:
Done